# Unit 3 Exercise 3, coefficient
Exercise has been downloaded and may contain alterations to the original code.

I've had to use a random cars file (cars3.csv) and pointed it to a github file on my repository.

Get the purpose of coefficient in the writing of the code. Reminder of differentiation from maths/economics when it came to linear regression prediction

In [16]:
import pandas as pd
from sklearn import linear_model

# Use the raw URL of the CSV file
df = pd.read_csv('https://raw.githubusercontent.com/cfinal15/cfinal15.github.io/main/Machine%20Learning/Unit%20Datasets/cars3.csv')

X = df[['Engine Information.Engine Statistics.Horsepower', 'Engine Information.Engine Statistics.Torque']]
y = df['Fuel Information.Highway mpg']

# Initialize and fit the linear regression model
regr = linear_model.LinearRegression()
regr.fit(X, y)

# Predict the CO2 emission for a car with given weight and volume
predictedCO2 = regr.predict([[2300, 1300]])

print(predictedCO2)

[3.01972408]


C:\Users\chris\AppData\Roaming\Python\Python311\site-packages\sklearn\base.py:493: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  warnings.warn(


# Coefficient

The coefficient is a factor that describes the relationship with an unknown variable.
In this case, we can ask for the coefficient value of weight against CO2, and for volume against CO2. The answer(s) we get tells us what would happen if we increase, or decrease, one of the independent values.

In [10]:
print(regr.coef_)

[0.00755095 0.00780526]


The result array represents the coefficient values of weight and volume.

Weight: 0.00755095
Volume: 0.00780526

These values tell us that if the weight increase by 1kg, the CO2 emission increases by 0.00755095g.

And if the engine size (Volume) increases by 1 cm3, the CO2 emission increases by 0.00780526 g.

I think that is a fair guess, but let test it!

We have already predicted that if a car with a 1300cm3 engine weighs 2300kg, the CO2 emission will be approximately 107g.

What if we increase the weight with 1000kg (from 2300 to 3300) what will be the CO2 emission?

Ans: 107.2087328 + (1000 * 0.00755095) = 114.75968

In [11]:
predictedCO2 = regr.predict([[3300, 1300]])

print(predictedCO2)

[114.75968007]
